crates/indexer/src/geyser/accounts/hpl_reward_center/purchase_ticket.rs

```diff
@@ -63,6 +67,49 @@ pub(crate) async fn process(
         .await
         .context("Failed to insert rewards purchase ticket")?;
 
+    client
+        .db()
+        .run({
+            let row = row.clone();
+            move |db| {
+                let rule = reward_rules::table
+                    .select(reward_rules::all_columns)
+                    .filter(
+                        reward_rules::reward_center_address.eq(row.clone().reward_center_address),
+                    )
+                    .first::<RewardRule>(db)
+                    .optional()?;
+
+                if let Some(r) = rule {
+                    let (buyer_reward, seller_reward) = calculate_payout(account_data.price, &r)?;
+
+                    let reward_payout = RewardPayout {
+                        purchase_ticket: Owned(key.to_string()),
+                        metadata: Owned(account_data.metadata.to_string()),
+                        reward_center: r.reward_center_address,
+                        buyer: Owned(account_data.buyer.to_string()),
+                        buyer_reward,
+                        seller: Owned(account_data.seller.to_string()),
+                        seller_reward,
+                        created_at: row.clone().created_at,
+                        slot: slot.try_into().unwrap_or_default(),
+                        write_version: write_version.try_into().unwrap_or_default(),
+                    };
+
+                    insert_into(reward_payouts::table)
+                        .values(&reward_payout)
+                        .on_conflict(reward_payouts::purchase_ticket)
+                        .do_update()
```

Comment:
This should be a do nothing so the payout rewards aren't recalculated in the case of the reward center reward rules being updated at a later time.